pandora-common/src/networking/directory_client.ts

```diff
@@ -101,6 +101,40 @@ export const ACCOUNT_SETTINGS_DEFAULT = Object.freeze<IDirectoryAccountSettings>
 	interfaceChatroomOfflineCharacterFilter: 'ghost',
 });
 
+export const DirectoryAccountSettingsLimitedSchema = z.object({
+	displayName: DisplayNameSchema,
+});
+export type IDirectoryAccountSettingsLimited = z.infer<typeof DirectoryAccountSettingsLimitedSchema>;
+
+type DirectoryAccountSettingsLimitedStoredSchemaAcc = {
+	[K in keyof IDirectoryAccountSettingsLimited]: ZodObject<{
+		value: typeof DirectoryAccountSettingsLimitedSchema.shape[K];
+		nextAllowedChange: ZodNumber;
+	}>;
+};
+export const DirectoryAccountSettingsLimitedStoredSchema = z.object(
+	KnownObject.entries(DirectoryAccountSettingsLimitedSchema.shape).reduce(
+		(acc, [key, value]) => {
+			acc[key] = z.object({
+				value,
+				nextAllowedChange: z.number(),
+			});
+			return acc;
+		},
+		// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+		{} as DirectoryAccountSettingsLimitedStoredSchemaAcc,
+	),
+);
+export type IDirectoryAccountSettingsLimitedStored = z.infer<typeof DirectoryAccountSettingsLimitedStoredSchema>;
+
+export const ACCOUNT_SETTINGS_LIMITED_STORED_DEFAULT = Object.freeze<IDirectoryAccountSettingsLimitedStored>({
+	displayName: { value: '', nextAllowedChange: 0 },
+});
+
+export const ACCOUNT_SETTINGS_LIMITED_LIMITS = Object.freeze<Readonly<Record<keyof IDirectoryAccountSettingsLimited, number>>>({
+	displayName: TimeSpanMs(1, 'months'),
```

Comment:
[discussion] I think allowing a change only once per month might be too high for this. How about once per week?
As I said before I honestly think the limiting is not needed at all. The action of changing your name has no performance cost for us and for users it will be similarly confusing if user changes the name once as as if they change it multiple times - they always see only the last value either way.
Only case this affects is when person A changes their name, person B sees it and _gets used to the new one_, and then A changes the name again. This will however be problem no matter what time period we choose. On other hand long period might as well be almost the same as not allowing the name change at all.